subprojects/scala/src/main/java/org/gradle/api/tasks/scala/ScalaCompile.java

```diff
@@ -54,6 +54,7 @@ public class ScalaCompile extends AbstractScalaCompile {
     public ScalaCompile() {
         super(new ScalaCompileOptions());
```

Comment:
To make `@Inject` work, we need to instantiate the options via Gradle. 
```suggestion
        super(getProject().getObjects().newInstance(ScalaCompileOptions.class))
```

Reply:
Aha ! I did try using `@Inject` but did not make this change. As you mentioned,  the code threw UnsupportedOperationException.

Will try this one out - Thanks!